src/clj/fluree/db/json_ld/commit.cljc

```diff
@@ -31,31 +28,30 @@
 
 (defn- subject-block-pred
   [db compact-fn list? p-flakes]
-  (go-try
-    (loop [[p-flake & r'] p-flakes
-           all-refs? nil
-           acc'      nil]
-      (let [pdt       (flake/dt p-flake)
-            ref?      (= const/$xsd:anyURI pdt)
-            [obj all-refs?] (if ref?
-                              [{"@id" (get-s-iri (flake/o p-flake)
-                                                 db compact-fn)}
-                               (if (nil? all-refs?) true all-refs?)]
-                              [{"@value" (-> p-flake
-                                             flake/o
-                                             (serde-json/serialize-object pdt))}
-                               false])
-            obj*      (cond-> obj
-                        list? (assoc :i (-> p-flake flake/m :i))
-                        (datatype/time-type? pdt)
-                        ;;need to retain the `@type` for times
-                        ;;so they will be coerced correctly when loading
-                        (assoc "@type"
-                               (get-s-iri pdt db compact-fn)))
-            next-acc' (conj acc' obj*)]
-        (if (seq r')
-          (recur r' all-refs? next-acc')
-          [next-acc' all-refs?])))))
+  (loop [[p-flake & r'] p-flakes
+         all-refs? nil
+         acc'      nil]
+    (let [pdt       (flake/dt p-flake)
+          ref?      (= const/$xsd:anyURI pdt)
+          [obj all-refs?] (if ref?
+                            [{"@id" (get-s-iri (flake/o p-flake)
+                                               db compact-fn)}
+                             (if (nil? all-refs?) true all-refs?)]
+                            [{"@value" (-> p-flake
+                                           flake/o
+                                           (serde-json/serialize-object pdt))}
+                             false])
+          obj*      (cond-> obj
+                      list? (assoc :i (-> p-flake flake/m :i))
+                      (datatype/time-type? pdt)
+                      ;;need to retain the `@type` for times
+                      ;;so they will be coerced correctly when loading
+                      (assoc "@type"
```

Comment:
This is super nitpick-y and totally up to you, but it took me a sec to understand what's going on here w/ the comment splitting the pred and the threaded form. Might be ever-so-slightly more readable to put them comment above the pred? Could even put a blank line above them for even more obvious grouping.